addon/parser.js

```diff
@@ -10,21 +10,31 @@ export default {
       rootField.selectionSet.push(field);
     });
 
-    model.eachRelationship((rel) => {
-      let relModel = store.modelFor(Ember.String.singularize(rel));
-      let field = new Type.Field(
-        rel,
-        null,
-        new Type.ArgumentSet(),
-        new Type.SelectionSet(new Type.Field('id'))
-      );
-
-      relModel.eachAttribute(function(attr) {
-        let relField = new Type.Field(attr);
-        field.selectionSet.push(relField);
-      });
+    model.eachRelationship((relName, {kind, type, options}) => {
+      if (options.async) {
```

Comment:
The new stuff here really blows up the inside of this loop to be quite complex. It had a bunch going on before, and now I find it pretty hard to part with the extract conditionals (this `if`, the ternaries and the `modelName !== relName && relName`). Could we decompose this a bit?